Add flag to suppress errors due to infeasible assumptions.

wuestholz opened the pull request:
This adds a flag to suppress errors that are currently reported in case the condition in a `klee_assume` is infeasible. This requires fewer solver queries than branching on the condition and terminating the branch where the condition does not hold (see https://github.com/klee/klee/commit/40d9c1d6296211f8127b34036509b225b491d4b8 for helper function that implements this).

delcypher commented:
@wuestholz Just to note 40d9c1d6296211f8127b34036509b225b491d4b8 is a bit of hack that will never end up in mainline KLEE. I was just doing it to run some experiments

wuestholz commented:
@delcypher Thank you for the feedback! Unfortunately, it seems that most of it got lost after I squashed the commits... Sorry, but I wasn't aware that this would happen.

I was assuming that https://github.com/klee/klee/commit/40d9c1d6296211f8127b34036509b225b491d4b8 was just a workaround. I mainly referred to it in order to demonstrate why built-in support for this is useful.

I hope I have addressed all of your feedback.

delcypher commented:
@wuestholz 
> Thank you for the feedback! Unfortunately, it seems that most of it got lost after I squashed the commits... Sorry, but I wasn't aware that this would happen.

Don't worry about it. You addressed the issues I raised well.

wuestholz commented:
@delcypher Thank you for the additional feedback! I have addressed all additional issues.

wuestholz commented:
@delcypher Is there anything else I should do before this can be merged?

ccadar commented:
Thanks, @wuestholz . This looks good to me, although I would perhaps change the help message to add "and exit silently instead", to make it clear that execution does not continue (e.g., by ignoring the assume). 

The only other thing to do is to document this new option at http://klee.github.io/docs/options/, via the https://github.com/klee/klee.github.io repository.

delcypher commented:
@wuestholz 
I was waiting for others to comment but I hadn't seen anything so I went ahead and merged (with a few very minor tweaks).

@ccadar 
I already add commits to master before seeing your comment. I've tried to address it in 48af356e766a73fe33cabbabc7c5edb263fb7d89

wuestholz commented:
@delcypher and @ccadar Thank you for merging this and the additional changes. I have created a PR (https://github.com/klee/klee.github.io/pull/18) that adds documentation for this flag to the website.
